dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -45,25 +45,31 @@ def __init__(self, options=None):
         :param options: Options for the numerical stats.
         :type options: NumericalOptions
         """
-        self.options = None
-        if options and isinstance(options, NumericalOptions):
-            self.options = options
+        if options and not isinstance(options, NumericalOptions):
+            raise ValueError("NumericalStatsMixin parameter 'options' must be "
+                             "of type NumericalOptions.")
         self.min = None
         self.max = None
         self.sum = 0
         self.variance = 0
         self.max_histogram_bin = 10000
+        self.min_histogram_bin = 1
         self.histogram_bin_method_names = [
             'auto', 'fd', 'doane', 'scott', 'rice', 'sturges', 'sqrt'
         ]
         self.histogram_selection = None
-        if options is not None and isinstance(options, NumericalOptions):
-            opt_method = options.histogram_and_quantiles.method
-            if isinstance(opt_method, str):
-                self.histogram_bin_method_names = [opt_method]
-                self.histogram_selection = opt_method
-            elif isinstance(opt_method, list):
-                self.histogram_bin_method_names = opt_method
+        self.user_set_histogram_bin = None
+        if options:
+            bin_count_or_method = \
+                options.histogram_and_quantiles.bin_count_or_method
+            if isinstance(bin_count_or_method, str):
+                self.histogram_bin_method_names = [bin_count_or_method]
+            elif isinstance(bin_count_or_method, list):
+                self.histogram_bin_method_names = bin_count_or_method
+            elif isinstance(bin_count_or_method, int):
+                self.user_set_histogram_bin = \
+                    options.histogram_and_quantiles.bin_count_or_method
```

Comment:
can be replaced by `bin_count_or_method ` to be consistent with the above

Reply:
fixing